jpype/_core.py

```diff
@@ -310,6 +312,17 @@ def shutdownJVM():
     _jpype.shutdown()
 
 
+# In order to shutdown cleanly we need the reference queue stopped
+# otherwise, we can experience a crash if a Java thread is waiting
+# for the GIL.
+def _JTerminate():
+    try:
+        _jpype.shutdown()
+    except RuntimeError:
+        pass
```

Comment:
exception silencing at this point might be the right thing to do. What kind of error do you expect here and would it be considerable to show them?

Reply:
The only case that yields a runtime exception here is that JVM is already shutdown.   Thus if the user has shutdown and then atexit does as well we get a unnecessary exception.   I considered just passing a flag to shutdown to make it conditional so that we don't have an exception thrown.   

If you would like me to write the alternative (cleaner) solution I can.

Reply:
There is one other obnoxious case that I have never resolved which is if someone calls shutdown from within a Proxy call back.   The whole shutdown pattern needs to reworked so that the JVM is in control and the master thread will wait for it to complete.   But that task has always been too much of a lift as it has to touch a lot of code, and the value of supporting it is low.